tests/components/vallox/test_config_flow.py

```diff
@@ -0,0 +1,298 @@
+    assert result["reason"] == "already_configured"
+
+
+async def test_import_with_custom_name(hass: HomeAssistant) -> None:
+    """Test that import is handled."""
+    name = "Vallox 90 MV"
+
+    with patch(
+        "homeassistant.components.vallox.config_flow.Vallox.get_info",
+        return_value=None,
+    ), patch(
+        "homeassistant.components.vallox.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data={"host": "1.2.3.4", "name": name},
+        )
+        await hass.async_block_till_done()
+
+    assert result["type"] == RESULT_TYPE_CREATE_ENTRY
+    assert result["title"] == name
+    assert result["data"] == {"host": "1.2.3.4", "name": "Vallox 90 MV"}
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_import_without_custom_name(hass: HomeAssistant) -> None:
+    """Test that import is handled."""
+    with patch(
+        "homeassistant.components.vallox.config_flow.Vallox.get_info",
+        return_value=None,
+    ), patch(
+        "homeassistant.components.vallox.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data={"host": "1.2.3.4"},
+        )
+        await hass.async_block_till_done()
+
+    assert result["type"] == RESULT_TYPE_CREATE_ENTRY
+    assert result["title"] == "Vallox"
+    assert result["data"] == {"host": "1.2.3.4", "name": "Vallox"}
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_import_invalid_ip(hass: HomeAssistant) -> None:
+    """Test that invalid IP error is handled during import."""
+    name = "Vallox 90 MV"
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_IMPORT},
+        data={"host": "vallox90mv.host.name", "name": name},
+    )
+    await hass.async_block_till_done()
+
+    assert result["type"] == RESULT_TYPE_ABORT
+    assert result["reason"] == "invalid_host"
+
+
+async def test_import_already_configured(hass: HomeAssistant) -> None:
+    """Test that an already configured Vallox device is handled during import."""
+    name = "Vallox 145 MV"
+
+    mock_entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={
+            CONF_HOST: "40.10.20.30",
+            CONF_NAME: "Vallox 145 MV",
+        },
+    )
+    mock_entry.add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_IMPORT},
+        data={"host": "40.10.20.30", "name": name},
+    )
+    await hass.async_block_till_done()
+
+    assert result["type"] == RESULT_TYPE_ABORT
+    assert result["reason"] == "already_configured"
+
+
+async def test_import_cannot_connect_OSError(hass: HomeAssistant) -> None:
```

Comment:
Please use lowercase snake_case function names.

Reply:
👍 

I thought about it too, but because no linter or pre-commit check complained I just assumed it was fine. Could/should it be enforced somehow? 🤔